### PR TITLE
[MINOR][DOCS] Fix various broken links and link anchors

### PR DESCRIPTION
**Pull request by nchammas**
### What changes were proposed in this pull request?

Fix various broken links and link anchors.

### Why are the changes needed?

Broken is broken, savvy?

### Does this PR introduce _any_ user-facing change?

Yes, it fixes broken links in user-facing documentation so they're no longer broken.

### How was this patch tested?

I built the docs and clicked on some of the fixed links to confirm they are no longer broken.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: docs/submitting-applications.md [hunk 1]
```diff
@@ -179,8 +179,7 @@ The master URL passed to Spark can be in one of the following formats:
 
 The `spark-submit` script can load default [Spark configuration values](configuration.html) from a
 properties file and pass them on to your application. By default, it will read options
-from `conf/spark-defaults.conf` in the Spark directory. For more detail, see the section on
-[loading default configurations](configuration.html#loading-default-configurations).
```
Comment (nchammas) [inline docs/submitting-applications.md:183]: This section doesn't exist anymore. I couldn't find something similar to link to, so I just deleted this sentence.

File: docs/spark-standalone.md [hunk 2]
```diff
@@ -751,8 +751,7 @@ Learn more about getting started with ZooKeeper [here](https://zookeeper.apache.
 
 **Configuration**
 
-In order to enable this recovery mode, you can set SPARK_DAEMON_JAVA_OPTS in spark-env by configuring `spark.deploy.recoveryMode` and related spark.deploy.zookeeper.* configurations.
-For more information about these configurations please refer to the [configuration doc](configuration.html#deploy)
```
Comment (nchammas) [inline docs/spark-standalone.md:755]: The `#deploy` section doesn't exist anymore. Neither does documentation for the `spark.deploy.*` configs mentioned here.

File: docs/submitting-applications.md [hunk 3]
```diff
@@ -179,8 +179,7 @@ The master URL passed to Spark can be in one of the following formats:
...
+from `conf/spark-defaults.conf` in the Spark directory.
```
Comment (yaooqinn) [inline docs/submitting-applications.md:182]:
```suggestion
from `conf/spark-defaults.conf` in the `SPARK_HOME` directory.
```

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks, merged to master

**Pull request closed without merging**